tests/prague/eip7623_increase_calldata_cost/test_transaction_validity.py

```diff
@@ -0,0 +1,524 @@
+    # intrinsic gas calculation, but we still test it to make sure that the transaction is
+    # correctly processed.
+    "blob_versioned_hashes",
+    [
+        pytest.param(
+            add_kzg_version(
+                [Hash(x) for x in range(1)],
+                EIP_4844_Spec.BLOB_COMMITMENT_VERSION_KZG,
+            ),
+            id="single_blob",
+        ),
+        pytest.param(
+            add_kzg_version(
+                [Hash(x) for x in range(6)],
+                EIP_4844_Spec.BLOB_COMMITMENT_VERSION_KZG,
+            ),
+            id="multiple_blobs",
+        ),
+    ],
+)
+@pytest.mark.parametrize(
+    "ty",
+    [pytest.param(3, id="type_3")],
+)
+def test_transaction_validity_type_3(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    tx: Transaction,
+) -> None:
+    """
+    Test transaction validity for transactions with access lists, blobs, but no contract creation.
+    """
+    state_test(
+        pre=pre,
+        post={},
+        tx=tx,
+    )
+
+
+@pytest.mark.parametrize(
+    "access_list",
+    [
+        pytest.param(
+            None,
+            id="no_access_list",
+        ),
+        pytest.param(
+            [AccessList(address=Address(1), storage_keys=[])],
+            id="single_access_list_no_storage_keys",
+        ),
+        pytest.param(
+            [AccessList(address=Address(1), storage_keys=[Hash(0)])],
+            id="single_access_list_single_storage_key",
+        ),
+        pytest.param(
+            [AccessList(address=Address(1), storage_keys=[Hash(k) for k in range(10)])],
+            id="single_access_list_multiple_storage_keys",
+        ),
+        pytest.param(
+            [AccessList(address=Address(a), storage_keys=[]) for a in range(10)],
+            id="multiple_access_lists_no_storage_keys",
+        ),
+        pytest.param(
+            [AccessList(address=Address(a), storage_keys=[Hash(0)]) for a in range(10)],
+            id="multiple_access_lists_single_storage_key",
+        ),
+        pytest.param(
+            [
+                AccessList(address=Address(a), storage_keys=[Hash(k) for k in range(10)])
+                for a in range(10)
+            ],
+            id="multiple_access_lists_multiple_storage_keys",
+        ),
+    ],
+)
+@pytest.mark.parametrize(
+    "authorization_list",
+    [
+        pytest.param(
+            [AuthorizationTuple(address=Address(1))],
+            id="single_authorization",
+        ),
+        pytest.param(
+            [AuthorizationTuple(address=Address(1)) for _ in range(10)],
+            id="multiple_authorizations",
+        ),
+    ],
+)
+@pytest.mark.parametrize(
+    "ty",
+    [pytest.param(4, id="type_4")],
+)
+def test_transaction_validity_type_4(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    tx: Transaction,
+) -> None:
+    """
+    Test transaction validity for transactions with access lists, authorization lists, but no
```

Comment:
legacy transactions and transactions with basefee but no accesslist
withdrawals?

Reply:
I think the gas refunds still need to be tested in a separate kind of test, one that verifies transaction execution and the deducted costs.

> legacy transactions

This is tested in `test_transaction_validity_type_0`

> transactions with basefee but no accesslist

Tested in `test_transaction_validity_type_1_type_2` with parametrization of `access_list=None` (see `id="no_access_list"`)

> withdrawals?

I think this is not an intrinsic transaction property but a block one.